src/utilities/delete-collection.tsx

```diff
@@ -0,0 +1,191 @@
+      ? delete_collection_permission && (
+          <DropdownItem
+            key='delete-collection-enabled'
+            onClick={() => onClick()}
+            data-cy='delete-collection-dropdown'
+          >
+            {t`Delete entire collection`}
+          </DropdownItem>
+        )
+      : delete_collection_permission && (
+          <Tooltip
+            key='delete-collection-disabled'
+            position='left'
+            content={
+              <Trans>
+                Cannot delete until collections <br />
+                that depend on this collection <br />
+                have been deleted.
+              </Trans>
+            }
+          >
+            <DropdownItem isDisabled>
+              {t`Delete entire collection`}
+            </DropdownItem>
+          </Tooltip>
+        );
+
+    if (!ret) {
+      ret = <div></div>;
+    }
+    return ret;
+  }
+
+  public deleteModal(
+    deleteCollection,
+    isDeletionPending,
+    confirmDelete,
+    collectionVersion,
+    cancelAction,
+    deleteAction,
+    onChange,
+  ) {
+    return (
+      deleteCollection && (
+        <DeleteModal
+          spinner={isDeletionPending}
+          cancelAction={() => cancelAction()}
+          deleteAction={() => deleteAction()}
+          isDisabled={!confirmDelete || isDeletionPending}
+          title={
+            collectionVersion
+              ? t`Delete collection version?`
+              : t`Delete collection?`
+          }
+        >
+          <>
+            <Text style={{ paddingBottom: 'var(--pf-global--spacer--md)' }}>
+              {collectionVersion ? (
+                <>
+                  {deleteCollection.all_versions.length === 1 ? (
+                    <Trans>
+                      Deleting{' '}
+                      <b>
+                        {deleteCollection.name} v{collectionVersion}
+                      </b>{' '}
+                      and its data will be lost and this will cause the entire
+                      collection to be deleted.
+                    </Trans>
+                  ) : (
+                    <Trans>
+                      Deleting{' '}
+                      <b>
+                        {deleteCollection.name} v{collectionVersion}
+                      </b>{' '}
+                      and its data will be lost.
+                    </Trans>
+                  )}
+                </>
+              ) : (
+                <Trans>
+                  Deleting <b>{deleteCollection.name}</b> and its data will be
+                  lost.
+                </Trans>
+              )}
+            </Text>
+            <Checkbox
+              isChecked={confirmDelete}
+              onChange={(val) => onChange(val)}
+              label={t`I understand that this action cannot be undone.`}
+              id='delete_confirm'
+            />
+          </>
+        </DeleteModal>
+      )
+    );
+  }
+
+  public deleteCollection(component, redirect, selectedRepo, addAlert) {
```

Comment:
might want to make this 

```suggestion
  public deleteCollection({ state, setState, load }, redirect, selectedRepo, addAlert) {
```

Reply:
I only slightly changed this to pass deleteCollection(state, setState, load, redirect...) without braces {}. If you insist on braces, tell me please why do you want that.

Reply:
This is fine :) the braces were there mostly for if we decided to only be explicit here, so we could still pass this or a object of that shape. Not needed when passing like that, so that's a detail now :)

Eventually we'll want an object instead of N params, to be able to add new (and named) fields, but that's a job for future us :)